bundles/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/internal/PentairControllerCircuit.java

```diff
@@ -0,0 +1,224 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.pentair.internal;
+
+import static org.openhab.binding.pentair.internal.PentairBindingConstants.*;
+
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * Class to manage Controller Circuits/Features
+ *
+ * @author Jeff James - initial contribution
+ *
+ */
+@NonNullByDefault
+public class PentairControllerCircuit {
+    //@formatter:off
+    public static final Map<Integer, String> CIRCUITNAME = MapUtils.mapOf(
```

Comment:
The reason I didn't make enums is that the strings have whitespaces so that they match the exact string in the Pentair controller menuing.  I also needed a reverse lookup for the items and so I thought just having a static hash map would be sufficient.  Since things are working now, unless you see a big issue here, i'd prefer to keep this code vs. spending more time on this.  Thoughts?

Reply:
Enums can take attributes, if you need the name in another notation. Reverse lookups can be implemented in an elegant way with streams. Your implementation irks me a bit, because you need to build an inverse map, do the cast. But I don't insist on changing it.

Reply:
Thanks for the pointers here - this helps in understanding Java.   I did end up converting to enums as a learning exercise so the next update this will be changed.